src/data/polynomial/ring_division.lean

```diff
@@ -678,6 +678,17 @@ begin
   exact hp (map_injective _ (no_zero_smul_divisors.algebra_map_injective T S) h)
 end
 
+lemma root_set_maps_to {p : T[X]} {S S'} [comm_ring S] [is_domain S] [algebra T S]
+  [comm_ring S'] [is_domain S'] [algebra T S'] (hp : p.map (algebra_map T S') ≠ 0)
+  (f : S →ₐ[T] S') : (p.root_set S).maps_to f (p.root_set S') :=
+λ x hx, begin
+  rw [mem_root_set_iff' hp, ← f.comp_algebra_map, ← map_map, eval_map],
+  erw [eval₂_hom, (mem_root_set_iff' (λ h, _) x).1 hx],
+  { apply _root_.map_zero },
+  rw [← f.comp_algebra_map, ← map_map, h] at hp,
+  exact hp (polynomial.map_zero f),
+end
```

Comment:
```suggestion
λ x, begin
  have H : map (algebra_map T S) p ≠ 0,
  { contrapose! hp,
    rw [← f.comp_algebra_map, ← map_map, hp, polynomial.map_zero] },
  simp_rw [mem_root_set_iff' hp, mem_root_set_iff' H, eval_map, ← alg_hom_eval₂_algebra_map],
  exact λ hx, hx.symm ▸ f.map_zero
end
```

It's possible that the `have` here should be a lemma? If we don't already have it (it seems simple!)

Reply:
Does `polynomial.map_ne_zero` help?